base_rest/components/service.py

```diff
@@ -91,114 +80,86 @@ def _log_call(self, func, params, secure_params, res):
             message = "REST call url %s method %s"
             _logger.debug(message, *args, extra=extra)
 
-    def _get_validator(self, validator_method):
-        if not hasattr(self, validator_method):
-            return None
-        v = getattr(self, validator_method)()
-        if isinstance(v, dict):
-            return Validator(v, purge_unknown=True)
-        return v
-
-    def _get_input_validator(self, method_name):
-        return self._get_validator("_validator_%s" % method_name)
-
-    def _get_output_validator(self, method_name):
-        return self._get_validator("_validator_return_%s" % method_name)
-
-    def _get_input_schema(self, method_name):
-        validator = self._get_input_validator(method_name)
-        if not validator:
-            return None
-        return validator.schema
-
-    def _get_output_schema(self, method_name):
-        validator = self._get_output_validator(method_name)
-        if not validator:
-            return None
-        return validator.schema
-
-    def _secure_input(self, method, params):
+    def _prepare_input_params(self, method, params):
```

Comment:
https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

for all the docstrings